Additional column info when creating table from recordset

cahilfoley opened the pull request:
What this does:
Currently the `identity` and `readOnly` properties are lost when converting a recordset to a table. This change passes those properties through to the `Table` instance which is useful when trying to perform bulk operations as writing to these columns will throw errors.

dhensby commented:
Nice, would it be possible to get a test for this, please?

willmorgan commented on lib/table.js:
```diff
@@ -33,6 +33,8 @@ function Table (name) {
       column.name = name
       column.nullable = options.nullable
       column.primary = options.primary
+      column.identity = options.identity
```
Should these be covered by `objectHasProperty`?

Should we also be looking at designating these info fields as column info and handling them in some array fashion?

```js
const additionalColumnInfoFields = ['nullable', 'primary', 'identity', 'readOnly', 'length'];
additionalColumnInfoFields.forEach((f) => {
    if (objectHasProperty(options, f)) {
        column[f] = options[f];
    }
}
```

cahilfoley replied:
From what I could see the `objectHasProperty` was only used for the `column.length` property because it was being set earlier in the function, only want to override that value of the `options` has something set. The rest didn't seem to exist anywhere at all.

I like your idea though, I was just trying to keep my changes to a minimum.

cahilfoley commented:
@dhensby just added a test for reading the values from some of the cases I was checking for, can you think of any other situations where this could come up?

dhensby commented:
Thanks @cahilfoley - I've pushed a couple changes to make the tests run (our test suite is crazily confusing, sorry). It appears to be failing on `msnodesqlv8` driver, maybe we don't get the same information back across the drivers, though that would be a shame.

If you don't have windows to test on I can look at debugging this a bit

cahilfoley commented:
Sorry @dhensby, having some problems even getting the full test suite running locally, unit tests are fine but I can't get the tedious ones running. I'll try spend some more time on it tomorrow or Saturday.

dhensby commented:
@cahilfoley any progress on this?

cahilfoley commented:
Apologies @dhensby, didn't have time to get this sorted for the project I was working on so ended up working around the issue. I should have some time this week to fix up the tests though!

cahilfoley commented:
@dhensby You were right that the msnodesqlv8 doesn't provide all of the fields - just `nullable` which I've added to be passed through. Updated the tests too to ignore the other `identity`/`readOnly` properties for msnodesqlv8 👍. Let me know what you think!

dhensby commented:
Thanks @cahilfoley, this is really good work and I appreciate the effort.

It may be worth us raising the lack of identity and readOnly column info issue with msnodesqlv8 and see if that can be added in the upstream driver.

Question now is, should this target 6.x or 7.x? 6.x is still the main one I'm focusing on and I have no immediate plans to release a 7.x stable so I think I'd prefer to target 6.x.

I'll have a go at rebasing this PR
